src/parser.py

```diff
@@ -133,7 +135,30 @@ def _parse_while_statement(self) -> WhileStatement:
 
     def _parse_function_statement(self) -> FunctionStatement:
         log.info(AppType.PARSER, 'Started parsing FunctionStatement')
-        pass
+
+        # name
+        name = self._consume_or_raise(IDENTIFIER_TOKENS, "Expected a function name")
+
+        # parameters
+        parameters = []
+        self._consume_or_raise(GROUP_OPENING_TOKENS, "Expected open paren after function name")
+        if not self._is_same_type(TokenType.RIGHT_PAREN):
+            parameters.append(self._consume_or_raise(IDENTIFIER_TOKENS, "Expected parameter name"))
+            while self._is_one_of_types(DELIMITER_TOKENS):
```

Comment:
Suggest refactoring similar to the call expression, since right now it will add one error message for any argument in excess of 254.

Reply:
This shouldn't matter since a function defined like this would always fail the arity check when called, but I made the change anyhow